### PR TITLE
adds cumulative sum for varmat

### PR DESCRIPTION
**Pull request by SteveBronder**
#### Submission Checklist

- [x] Run unit tests
- Documentation
    - [ ] If a user-facing facing change was made, the documentation PR is here: <LINK>
    - [x] OR, no user-facing changes were made

## Release notes

Adds `cumulative_sum(Vector/RowVector)` for `var<Matrix>` types

## Copyright and Licensing

By submitting this pull request, the copyright holder is agreeing to 
license the submitted work under the BSD 3-clause license (https://opensource.org/licenses/BSD-3-Clause)

### CODE REVIEW

**Review by WardBrian — APPROVED** [review]

**Comment by SteveBronder** [comment]
I'm going to wait until we do that patch release to merge

**Pull request merged**
